### PR TITLE
feat: ConfigProvider support DatePicker & TimePicker className and style properties

### LINKED ISSUE #43051: ConfigProvider support component className and style
## Summary
`ConfigProvider` adds new props, which are used to add custom `className` and `style` to components globally.

## Motivation
When customizing components, some custom `className` and `style` are often added to the component. In the past, it was recommended to repackage the components of `antd`, but this would cause too much mental burden on users, such as forcing developers to import from custom components, etc., and it would be redundant if it was a simple transformation. Remain. Therefore, we provide properties on `ConfigProvider` that can globally assign `className` and `style` to components to facilitate custom styles.

## API
ConfigProvider

| Property | Description | Type | Default |
| --- | --- | --- | --- |
| button | Set Button common props | `{ className?: string, style?: React.CSSProperties }` | - |
			
## Example PR

- #42748

## RFC

- #39862

## Task List

- [x] Button
- [x] Checkbox @MuxinFeng #43107
- [x] Descriptions @MuxinFeng #43106
- [x] Divider @MuxinFeng  #43042
- [x] Empty @MuxinFeng #43061
- [x] Radio @MuxinFeng #43067
- [x] Layout @MuxinFeng #43241
- [x] Form @MuxinFeng #43230
- [x] Input @MuxinFeng #43227
- [x] Mentions @MuxinFeng #43192
- [x] Select @MuxinFeng #43217
- [x] Cascader @MuxinFeng #43205
- [x] Anchor @MuxinFeng #43189
- [x] Modal @MuxinFeng #43219
- [x] Progress @Yuiai01 #43340
- [x] Menu @Yuiai01 #43382
- [x] Table @li-jia-nan #43301
- [x] Tag @li-jia-nan #43294
- [x] TimePicker @Yuiai01 #43418
- [x] DatePicker @Yuiai01 #43418
- [x] Calendar @Yuiai01 #43385
- [x] Carousel @Yuiai01 #43412
- [x] Badge @li-jia-nan #43197
- [x] Rate @li-jia-nan #43270
- [x] Card @li-jia-nan #43308
- [x] Tabs @li-jia-nan #43319
- [x] Avatar @li-jia-nan #43315
- [x] Switch @li-jia-nan #43283
- [x] Pagination @li-jia-nan #43100
- [x] Breadcrumb @li-jia-nan #43095
- [x] Slider @li-jia-nan #43092
- [x] Tree @Yuiai01 #43060
- [x] Collapse @Yuiai01 #43390
- [x] Skeleton @Yuiai01 #43062
- [x] Transfer @Yuiai01 #43063
- [x] Message @li-jia-nan #43333
- [x] Alert @Yuiai01 #43384
- [x] List @Yuiai01 #43387
- [x] Statistic @Yuiai01 #43408
- [x] Drawer @Yuiai01 #43396
- [x] Timeline @Yuiai01 #43392
- [x] Typography @li-jia-nan #43070
- [x] Upload @li-jia-nan #43325
- [x] Steps @li-jia-nan #43073
- [x] Notification @li-jia-nan #43328
- [x] Result @li-jia-nan #43080
- [x] Image @li-jia-nan #43074
- [x] Segmented @li-jia-nan #43072
- [x] Spin @li-jia-nan #43071
- [x] Space @RedJue  #42748
- [x] ColorPicker @li-jia-nan #43339

### PR DESCRIPTION
**Pull request by Yuiai01**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- close #43051
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | ConfigProvider support DatePicker & TimePicker className and style properties          |
| 🇨🇳 Chinese |  ConfigProvider 支持 DatePicker 和 TimePicker 组件的 className 和 style 属性         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at e523eb7</samp>

This pull request adds two new props, `timePicker` and `datePicker`, to the `ConfigProvider` component, which allow users to customize the style and className of the `TimePicker` and `DatePicker` components globally. It updates the documentation, the demo files, and the internal logic of the `generatePicker` functions to support these new props.

### <samp>🤖 Generated by Copilot at e523eb7</samp>

*  Add `timePicker` and `datePicker` props to `ConfigProvider` component to allow setting common style and className props for `TimePicker` and `DatePicker` components ([link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaL135-R140), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56L179-R184), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56L311-R318), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56L405-R414))
*  Update `ConfigProvider` documentation in English and Chinese to describe the new props and their usage ([link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-975af6d8b4e359c4e88a7586cf1c91b35989cdadc9f2dc9ca16a376b5f5b0563R116), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-975af6d8b4e359c4e88a7586cf1c91b35989cdadc9f2dc9ca16a376b5f5b0563R148), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-b6a7375df96aaccf21f7452705af80f8cd8c7323f0fc2b9dbdc71500b8478775R118), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-b6a7375df96aaccf21f7452705af80f8cd8c7323f0fc2b9dbdc71500b8478775R150))
*  Modify `generateSinglePicker` and `generateRangePicker` functions to consume the style and className props from `ConfigProvider` and apply them to the `Picker` and `RCRangePicker` components ([link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-678a0178404ab41221775c01d203e79ccb84496e2f0b7cf4ef38222d7a7b562bR26-R27), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-678a0178404ab41221775c01d203e79ccb84496e2f0b7cf4ef38222d7a7b562bL39-R43), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-678a0178404ab41221775c01d203e79ccb84496e2f0b7cf4ef38222d7a7b562bR49), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-678a0178404ab41221775c01d203e79ccb84496e2f0b7cf4ef38222d7a7b562bL58-R71), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-678a0178404ab41221775c01d203e79ccb84496e2f0b7cf4ef38222d7a7b562bL156-R173), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-678a0178404ab41221775c01d203e79ccb84496e2f0b7cf4ef38222d7a7b562bL186-R201), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-fba17c237c150271f91ee2ca848654484ecbbbcc71b010a8947620e945e10e34R44), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-fba17c237c150271f91ee2ca848654484ecbbbcc71b010a8947620e945e10e34L56-R57), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-fba17c237c150271f91ee2ca848654484ecbbbcc71b010a8947620e945e10e34L136-R140))
*  Add `ConfigProvider` component to `TimePicker` and `DatePicker` demo files and pass some style and className props to them ([link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-4ab01e142168d67b52a674b8de11e721f5a69f69c11b299eb26c775848ff0b99L2-R2), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-4ab01e142168d67b52a674b8de11e721f5a69f69c11b299eb26c775848ff0b99L10-R14), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-5d7175667bac480675976be1ec06b3e8c89c4919e69bdcc607e08ee591cd51bcL1-R5), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-5d7175667bac480675976be1ec06b3e8c89c4919e69bdcc607e08ee591cd51bcL14-R16))
*  Add test cases to verify that `TimePicker` and `DatePicker` components can receive the style and className props from `ConfigProvider` and that they are applied correctly to the DOM elements ([link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-58612c0e42c2460919a2c73d9dac7eed9d88cb7d97da183b5f94e83ce74b8149R15), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-58612c0e42c2460919a2c73d9dac7eed9d88cb7d97da183b5f94e83ce74b8149R47), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-58612c0e42c2460919a2c73d9dac7eed9d88cb7d97da183b5f94e83ce74b8149R950-R977), [link](https://github.com/ant-design/ant-design/pull/43418/files?diff=unified&w=0#diff-58612c0e42c2460919a2c73d9dac7eed9d88cb7d97da183b5f94e83ce74b8149R1173-R1200))

### CODE REVIEW

File: components/date-picker/generatePicker/generateSinglePicker.tsx [hunk 1]
```diff
@@ -55,7 +60,15 @@ export default function generatePicker<DateType>(generateConfig: GenerateConfig<
...
-        const { getPrefixCls, direction, getPopupContainer } = useContext(ConfigContext);
+        const {
+          getPrefixCls,
+          direction,
+          getPopupContainer,
+          // Consume different styles according to different names
+          [customConsumerName]: consumerStyle,
+        } = useContext(ConfigContext);
+
+        console.log(consumerName);
```
Comment (li-jia-nan) [inline components/date-picker/generatePicker/generateSinglePicker.tsx:71]: console 没删

Comment (Yuiai01) [reply components/date-picker/generatePicker/generateSinglePicker.tsx:71]: 小弟这就去删掉

File: components/date-picker/generatePicker/generateRangePicker.tsx [hunk 2]
```diff
@@ -53,7 +54,7 @@ export default function generateRangePicker<DateType>(generateConfig: GenerateCo
     } = props;
 
     const innerRef = React.useRef<RCRangePicker<DateType>>(null);
-    const { getPrefixCls, direction, getPopupContainer } = useContext(ConfigContext);
+    const { getPrefixCls, direction, getPopupContainer, datePicker } = useContext(ConfigContext);
```
Comment (MadCcc) [inline components/date-picker/generatePicker/generateRangePicker.tsx:57]: RangePicker 应该和 DatePicker 的 style 无关了，这里可以先不加

Comment (Yuiai01) [reply components/date-picker/generatePicker/generateRangePicker.tsx:57]: 好

File: components/date-picker/generatePicker/generateSinglePicker.tsx [hunk 3]
```diff
@@ -36,12 +38,15 @@ export default function generatePicker<DateType>(generateConfig: GenerateConfig<
   function getPicker<InnerPickerProps extends DatePickerProps>(
     picker?: PickerMode,
     displayName?: string,
+    consumerName?: StyleConsumerName,
```
Comment (MadCcc) [inline components/date-picker/generatePicker/generateSinglePicker.tsx:41]: 应该可以用 DisplayName 判断一下吧，就不用加新参数了

Comment (Yuiai01) [reply components/date-picker/generatePicker/generateSinglePicker.tsx]: 嗯嗯，已更改～

**Review by MadCcc — APPROVED** [review]

**Pull request merged**